xdsl/dialects/linalg.py

```diff
@@ -103,6 +105,52 @@ def __init__(
             regions=[body],
         )
 
+    def get_indexing_maps(self) -> list[AffineMap]:
+        return [attr.data for attr in self.indexing_maps]
+
+    def get_num_loops(self) -> int:
+        return self.indexing_maps.data[0].data.num_dims
+
+    def get_loops_to_shapes_map(self) -> AffineMap:
+        result_exprs = [res for map in self.get_indexing_maps() for res in map.results]
+
+        dims = self.get_num_loops()
+
+        # FIXME: Support symbols.
+        for map in self.get_indexing_maps():
+            if map.num_symbols != 0:
+                raise NotImplementedError(
+                    "Indexing maps with symbols not supported for now."
+                )
+
+        syms = 0
+        return AffineMap(dims, syms, result_exprs)
+
+    def get_shapes_to_loops_map(self) -> AffineMap:
+        loops_to_shapes = self.get_loops_to_shapes_map()
+        inverse = loops_to_shapes.inverse_permutation()
+        if not inverse:
+            raise NotImplementedError(
+                "Non-invertible maps need dynamic shapes, which are not implemented."
+            )
+        return inverse
+
+    def get_static_shapes(self) -> list[int]:
+        sizes: list[int] = []
+        for input in self.inputs:
+            if isinstance(input.type, ShapedType):
+                for shape in input.type.get_shape():
```

Comment:
```suggestion
                for dim in input.type.get_shape():
```
Aren't elements of shapes dimensions?